src/internal.rs

```diff
@@ -430,12 +430,12 @@ pub(crate) struct Rav1dFrameContext_task_thread {
     pub init_task: Rav1dTask,
     pub num_tasks: c_int,
     pub num_tile_tasks: c_int,
-    pub init_done: atomic_int,
-    pub done: [atomic_int; 2],
+    pub init_done: AtomicI32,
+    pub done: [AtomicI32; 2],
     pub retval: Rav1dResult,
     pub update_set: bool, // whether we need to update CDF reference
-    pub error: atomic_int,
-    pub task_counter: atomic_int,
+    pub error: AtomicI32,
```

Comment:
I think we want this to be an `Atomic<Rav1dError>`.  We can `#[derive(Atom)]` on `enum Rav1dError` (using `atomig`).

Reply:
It might be fine to just do that later, though.

Reply:
Definitely at some point, but that isn't the priority now or in the near future.